### ISSUE TITLE
Global Media Controls crashing Chromium when used

### ISSUE THREAD

**Issue by KJP12** [ungoogled-software/ungoogled-chromium-archlinux#185]
## Happens in Upstream Ungoogled Chromium?
No, Ungoogled Chromium `101.0.4951.64 (Official Build, ungoogled-chromium) (64-bit)` from [Third Party Binaries](https://ungoogled-software.github.io/ungoogled-chromium-binaries/releases/linux_portable/64bit/101.0.4951.64-1) does not crash when Global Media Controls is used.

## Steps to Reproduce
- Have Ungoogled Chromium from OpenSUSE's build servers as setup in this repo's README.md.
  - Note that flags nor extensions, aside from `--disable-features=GlobalMediaControls` does not change the end result, aside from making it impossible to attempt to reproduce by omitting the button outright.
- Have any audio source playing.
  - All crashes so far has been using YouTube in specific, but other sources may suffice.
- Click the global media controls.

## chrome://version
<details><summary>Arch Linux (Note: Launched via <code>chromium --user-data-dir=/tmp/chromium</code>)</summary>

Chromium | 101.0.4951.64 (Official Build, ungoogled-chromium) Arch Linux (64-bit)
-- | --
Revision | d1daa9897e1bc1d507d6be8f2346e377e5505905-refs/branch-heads/4951@{#1208}
OS | Linux
JavaScript | V8 10.1.124.12
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36
Command Line | /usr/lib/chromium/chromium --user-data-dir=/tmp/chromium --enable-crashpad --flag-switches-begin --flag-switches-end
Executable Path | /usr/lib/chromium/chromium
Profile Path | /tmp/chromium/Default
Variations | 8d569531-4f54ac4d

</details>

<details><summary>Third Party Build (reference)</summary>

As for some unexplainable reason, GitHub isn't wanting to render the entry line for `Chromium`: `101.0.4951.64 (Official Build, ungoogled-chromium) (64-bit)`

Variations raw, as it also got crushed:

8d569531-4f54ac4d
7beeeebd-70ea8f25
41ec777c-70ea8f25

Chromium | 101.0.4951.64 (Official Build, ungoogled-chromium) (64-bit)
-- | --
Revision | d1daa9897e1bc1d507d6be8f2346e377e5505905-refs/branch-heads/4951@{#1208}
OS | Linux
JavaScript | V8 10.1.124.12
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36
Command Line | /home/kjp12/Downloads/ungoogled-chromium_101.0.4951.64-1.1_linux/chrome --enable-crashpad --flag-switches-begin --bookmark-bar-ntp=never --blink-settings=disallowFetchForDocWrittenScriptsInMainFrame=true --extension-mime-request-handling=always-prompt-for-install --keep-old-history --ozone-platform-hint=auto --scroll-tabs=always --show-avatar-button=incognito-and-guest --enable-features=CopyLinkToText,GlobalMediaControlsModernUI,OverlayScrollbar,ParallelDownloading,PlaybackSpeedButton,ReaderMode,TabSearchMediaTabs:Also+show+Media+Tabs+in+Open+Tabs+Section/true,UseDownloadOfflineContentProvider,VaapiVideoDecoder,WebContentsForceDark:inversion_method/cielab_based/image_behavior/none/foreground_lightness_threshold/150/background_lightness_threshold/205,WebRtcEnableCaptureMultiChannelApm,WebRtcHideLocalIpsWithMdns,WebRtcHybridAgc --disable-features=ClearDataOnExit,LiveCaption,MediaSessionWebRTC,WebRtcAnalogAgcClippingControl --flag-switches-end --ozone-platform=wayland
Executable Path | /home/kjp12/Downloads/ungoogled-chromium_101.0.4951.64-1.1_linux/chrome
Profile Path | /home/kjp12/.config/chromium/Default
Variations | 8d569531-4f54ac4d7beeeebd-70ea8f2541ec777c-70ea8f25

</details>

## Environment
KDE Plasma 5.24.5
KDE Frameworks 5.94.0
Qt Version: 5.15.4
GTK3 Version: 3.24.33
Kernel: Linux 5.17.7-zen1-1-zen (Note: also occured on LTS)
Graphics Platform: Wayland (Note: also occurs under X11 and XWayland)

**Comment by jstkdng** [comment]
Some screenshots would be nice

**Comment by KJP12** [comment]
![image](https://user-images.githubusercontent.com/8528256/169634002-16d929c8-1a40-4193-8aca-d7e5817819a7.png)
This is the global media controls button I'm talking about.

Unsure what else to take screenshots of, as there isn't much more *to* screenshot, as the button results in a crash, even if it is visually fine for the short moment it's open.

**Comment by jstkdng** [comment]
I'm not experiencing this issue.
![image](https://user-images.githubusercontent.com/52050345/169637121-8e40c6be-7c98-49a1-80e8-18466b1c34b7.png)
I can pause and unpause if required, ofc, I rarely use this feature most of the time. I'm currently running sway.
What flags are you currently using? Maybe one of them are causing the issue.
I have these flags at the moment.
```
--extension-mime-request-handling=always-prompt-for-install
--force-dark-mode
--enable-features=WebUIDarkMode,OverlayScrollbar,VaapiVideoDecoder
--disable-features=UseChromeOSDirectVideoDecoder,SidePanel
--pasword-store=gnome
--ozone-platform-hint=auto
--disable-gpu-memory-buffer-compositor-resources
--disable-gpu-memory-buffer-video-frames
--enable-hardware-overlays
--gtk-version=4
```

**Comment by emansom** [comment]
Are you running Sway with a local dbus session? Chromium talks [`MPRIS`](https://wiki.archlinux.org/title/MPRIS) behind the scene, and this requires a active local user scoped D-Bus session.

Works without issue here on GNOME.

**Comment by KJP12** [comment]
> I'm currently running sway.

I'm currently using KDE Plasma, so, that maybe a possible point where a difference is showing through.

> What flags are you currently using? Maybe one of them are causing the issue.

Both these sets of flags caused issues when used with the Arch distribution of Chromium:

```
--user-data-dir=/tmp/chromium
--enable-crashpad
--flag-switches-begin
--flag-switches-end
```

```
--force-dark-mode
--enable-crashpad
--flag-switches-begin
--bookmark-bar-ntp=never
--blink-settings=disallowFetchForDocWrittenScriptsInMainFrame=true
--extension-mime-request-handling=always-prompt-for-install
--keep-old-history
--ozone-platform-hint=auto
--scroll-tabs=always
--show-avatar-button=incognito-and-guest
--enable-features=CopyLinkToText,GlobalMediaControlsModernUI,OverlayScrollbar,ParallelDownloading,PlaybackSpeedButton,ReaderMode,TabSearchMediaTabs:Also+show+Media+Tabs+in+Open+Tabs+Section/true,UseDownloadOfflineContentProvider,VaapiVideoDecoder,WebContentsForceDark:inversion_method/cielab_based/image_behavior/none/foreground_lightness_threshold/150/background_lightness_threshold/205,WebRtcEnableCaptureMultiChannelApm,WebRtcHideLocalIpsWithMdns,WebRtcHybridAgc
--disable-features=ClearDataOnExit,LiveCaption,MediaSessionWebRTC,WebRtcAnalogAgcClippingControl --flag-switches-end
--ozone-platform=wayland
```

Stock profile (set to `/tmp/chromium` to get a sufficiently clean profile), my usual profile and anywhere between has no difference, and is guaraneed to crash on interacting with or closing the global media controls.


> Are you running Sway with a local dbus session? Chromium talks [`MPRIS`](https://wiki.archlinux.org/title/MPRIS) behind the scene, and this requires a active local user scoped D-Bus session.

KDE launches its own D-Bus session and passes it through to Chromium as expected. Media controls via D-Bus works fine on both the Arch Linux distribution and the generic build. Although, this doesn't seem like it'd be related to the global media controls model crashing the browser.

**Comment by networkException** [comment]
I missed a patch getting added in the arch packaging, this will be fixed by 102.0.5005.115-2
